File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    int total = 40;
    // TODO: define the 'expectedMinutesInOven()' method
    public int expectedMinutesInOven(){
        return total;
    }
    // TODO: define the 'remainingMinutesInOven()' method
    public int remainingMinutesInOven(int actual){
        return expectedMinutesInOven() - actual;
    }
    // TODO: define the 'preparationTimeInMinutes()' method
    public int preparationTimeInMinutes(int layers) {
        return layers * 2;
    }
    // TODO: define the 'totalTimeInMinutes()' method
    public int totalTimeInMinutes(int layers, int actual){
        return preparationTimeInMinutes(layers) + actual;
    }
}

File: solutions/r/lasagna/1/lasagna.R
# TODO: define the 'expected_minutes_in_oven()' function
expected_minutes_in_oven <- function(){
  60
}
# TODO: define the 'remaining_time_in_minutes()' function
remaining_time_in_minutes <- function(x){
  expected_minutes_in_oven() - x
}
# TODO: define the 'prep_time_in_minutes()' function
prep_time_in_minutes <- function(x){
  x * 2
}
# TODO: define the 'elapsed_time_in_minutes()' function
elapsed_time_in_minutes <- function(x, y) {
  prep_time_in_minutes(x) + y
}